pkg/cfn/manager/api.go

```diff
@@ -178,8 +178,23 @@ func (c *StackCollection) CreateStack(ctx context.Context, stackName string, res
 
 // createClusterStack creates the cluster stack
 func (c *StackCollection) createClusterStack(ctx context.Context, stackName string, resourceSet builder.ResourceSetReader, errCh chan error) error {
+	// Do a preliminary check, so we don't end up waiting for nothing.
+	stack, err := c.DescribeStack(ctx, &types.Stack{
+		StackName: &stackName,
+	})
+	if err != nil && !IsStackDoesNotExistError(err) {
+		return err
+	}
+	if stack != nil {
+		switch stack.StackStatus {
+		case types.StackStatusCreateComplete, types.StackStatusDeleteComplete:
+			close(errCh) // no need to wait on this task anymore
+			return nil
```

Comment:
instead of `return nil`, should this not be a descriptive error about why it's returning early, which would be passed to the `errCh`? maybe there's something that I'm not seeing 🤔

Reply:
There is no error here. We got what we were looking for.

Reply:
To be more clear. This check if the stack has the desired status, which is `CREATE_COMPLETE` or `DELETE_COMPLETE` which the below check will be looking for as well. This is to avoid having to wait a minute until the next check if the stack already has the state we want anyways.

Reply:
I saw that, but at this point the stack has not been created yet. If a stack is found with `CREATE_COMPLETE`, couldn't it be a totally different one from a different operation?

The check below checks that the stack that was just created by `c.createStackRequest` was properly created, which is different.